src/popup/components/Header.vue

```diff
@@ -107,14 +107,18 @@ export default defineComponent({
     );
 
     function back() {
-      if (root.$route.meta?.backRoute) {
+      const { fullPath, meta } = root.$route;
+
+      if (meta?.directBackRoute) {
+        return root.$router.go(-1);
+      }
+      if (meta?.backRoute) {
         // TODO: rewrite back button logic in more unified way
-        return root.$router.push(root.$route.meta?.backRoute);
+        return root.$router.push(meta?.backRoute);
       }
-      let { fullPath } = root.$route;
-      fullPath = fullPath.endsWith('/') ? fullPath.slice(0, -1) : fullPath;
+      const path = fullPath.endsWith('/') ? fullPath.slice(0, -1) : fullPath;
       return root.$router.push(
-        fullPath.substr(0, fullPath.lastIndexOf('/')) || { name: currentHomeRouteName.value },
+        path.substr(0, path.lastIndexOf('/')) || { name: currentHomeRouteName.value },
```

Comment:
Ternary and '||" operators are used sequentially. Decreases the code readability.

Could be rewritten to
```js
    const path = fullPath.endsWith('/') ? fullPath.slice(0, -1) : fullPath;
    const lastSlashIndex = path.lastIndexOf('/');
    if (lastSlashIndex !== -1) {
      path = path.substr(0, lastSlashIndex);
    } else {
      path = { name: currentHomeRouteName };
    }
    return root.$router.push(path);
```